hal/src/nRF52840/ble_hal.cpp

```diff
@@ -15,739 +15,3147 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
+#include "logging.h"
+LOG_SOURCE_CATEGORY("hal.ble")
+
 #include "ble_hal.h"
 
-#include "nrf_ble_gatt.h"
-#include "ble_advertising.h"
-#include "ble_conn_params.h"
-#include "ble.h"
+#if HAL_PLATFORM_BLE
 
+/* Headers included from nRF5_SDK/components/softdevice/s140/headers */
+#include "ble.h"
+#include "ble_gap.h"
+#include "ble_gatt.h"
+#include "ble_gatts.h"
+#include "ble_gattc.h"
+#include "ble_types.h"
+/* Headers included from nRF5_SDK/components/softdevice/common */
 #include "nrf_sdh.h"
 #include "nrf_sdh_ble.h"
 #include "nrf_sdh_soc.h"
 
-#include "logging.h"
+#include "app_util.h"
+
+#include "concurrent_hal.h"
+#include "static_recursive_mutex.h"
+#include <mutex>
+
+#include "gpio_hal.h"
+#include "device_code.h"
+#include "system_error.h"
+#include "sdk_config_system.h"
+#include "spark_wiring_vector.h"
+#include "simple_pool_allocator.h"
+#include <string.h>
+#include <memory>
+
+static_assert(NRF_SDH_BLE_PERIPHERAL_LINK_COUNT == 1, "Multiple simultaneous peripheral connections are not supported");
+static_assert(NRF_SDH_BLE_TOTAL_LINK_COUNT <= 20, "Maximum supported number of concurrent connections in the peripheral and central roles combined exceeded");
+
+using spark::Vector;
+
+#define BLE_CONN_CFG_TAG 1
+
+#define NRF_CHECK_RETURN(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != NRF_SUCCESS) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+                return sysError(_ret); \
+            } \
+        } while (false)
+
+#define NRF_CHECK_RETURN_VOID(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != NRF_SUCCESS) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+            } \
+        } while (false)
+
+#define CHECK_RETURN(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != SYSTEM_ERROR_NONE) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+                return _ret; \
+            } \
+        } while (false)
 
-LOG_SOURCE_CATEGORY("hal.ble")
 
+#define CHECK_RETURN_VOID(ret, str) \
+        do { \
+            const int _ret = ret; \
+            const char* _str = str; \
+            if (_ret != SYSTEM_ERROR_NONE) { \
+                if (_str != nullptr) { \
+                    LOG(ERROR, "%s failed: %u", _str, _ret); \
+                } \
+                return; \
+            } \
+        } while (false)
+
+
+StaticRecursiveMutex s_bleMutex;
+class bleLock {
```

Comment:
Nitpicking: classes are PascalCase, so `BleLock`. It's also probably better not to use `std::lock_guard` and implement the full lock class, see `ThreadLock` in `ot_api.h`.